http-clients/apache5-client/src/main/java/software/amazon/awssdk/http/apache5/Apache5HttpClient.java

```diff
@@ -596,12 +606,12 @@ public void setDnsResolver(DnsResolver dnsResolver) {
         }
 
         @Override
-        public Builder socketFactory(ConnectionSocketFactory socketFactory) {
+        public Builder socketFactory(SSLConnectionSocketFactory socketFactory) {
```

Comment:
Is there any reason we are changing this?

Reply:
https://github.com/apache/httpcomponents-client/blob/4021b7c47914efaf9cf30b8d00bbd129295c0639/httpclient5/src/main/java/org/apache/hc/client5/http/impl/io/PoolingHttpClientConnectionManagerBuilder.java#L122-L127


    public final PoolingHttpClientConnectionManagerBuilder setSSLSocketFactory(
            final org.apache.hc.client5.http.socket.LayeredConnectionSocketFactory sslSocketFactory) {
        this.tlsSocketStrategy = (socket, target, port, attachment, context) ->
                (SSLSocket) sslSocketFactory.createLayeredSocket(socket, target, port, context);
        return this;
    }


PoolingHttpClientConnectionManagerBuilder only accepts  LayeredConnectionSocketFactory thus we need to do it.

Reply:
Should we use LayeredConnectionSocketFactory instead?

Reply:
Discussed offline, it's deprecated. Will be addressed in further PR.